pkg/jobqueue/jobqueue.go

```diff
@@ -67,7 +67,23 @@ type JobQueue interface {
 	// finished, respectively.
 	//
 	// Lastly, the IDs of the jobs dependencies are returned.
+	// Deprecated : JobStatus, uses too much bandwidth, prefer JobStatusWoResult instead
 	JobStatus(id uuid.UUID) (jobType string, channel string, result json.RawMessage, queued, started, finished time.Time, canceled bool, deps []uuid.UUID, dependents []uuid.UUID, err error)
+	// Does the same as JobStatus but without fetching the result blob
+	// To be used instead of JobStatus if possible
+	JobStatusWoResult(id uuid.UUID) (jobType string, channel string, queued, started, finished time.Time, canceled bool, deps []uuid.UUID, dependents []uuid.UUID, err error)
+
+	// Query a specific path in a JSON field.
+	// field -> the name of the JSON field in the DB
+	// path -> a dot separated path within the field, for example osbuild_output.error
+	// response -> a pointer to the data container to Unmarshal to
+	// Returns an Unmarshalled response
+	QueryJsonField(id uuid.UUID, field string, path string, response any) error
```

Comment:
I'm not sure if `field` is the best name. In the json blob it's field, but in the psql jobqueue it's a column.

Also, if we're querying multiple paths at the same time, could it be better to accept an array of `field:paths` ?

Reply:
Actually, can we accept `path` only and just always query the result column? As we only accept querying the result as a json blob.

Reply:
Oh yes, multiple queries at the same time, that's a good idea! I'll try to figure out something for this.

Reply:
I haven't setup the multiple queries yet. But I removed the `field` argument for the function, because as you said it's quite useless. And I renamed the function accordingly, so it's not QueryJsonField, but QueryResultFied as it's always going to fetch a field withing the `result` column.

Reply:
I'm resolving this comment because everything has changed since. But multiple query in the same time are a thing now :)